op-service/bsc/compat.go

```diff
@@ -14,6 +16,25 @@ var DefaultBaseFee = big.NewInt(3000000000)
 var DefaultOPBNBTestnetBaseFee = big.NewInt(5000000000)
 var OPBNBTestnet = big.NewInt(5611)
 
+const (
+	percentile     = 50
+	CountBlockSize = 21
+	// BlockInfoCacheCap 21 validators in bsc, finalize block requires 15, 2 * (21+15) = 72 to buffer
+	BlockInfoCacheCap = 72
+)
+
+type BlockInfo struct {
+	BlockHash      common.Hash
+	ParentHash     common.Hash
+	MedianGasPrice *big.Int
+}
+
+var BlockInfoCache *lru.Cache[common.Hash, BlockInfo]
```

Comment:
Without this cache, we need to calculate the median of each block of 21 blocks each time. If the traffic is high, there may be thousands of transactions in each block, may affect block production efficiency.

Reply:
I think it's better to leave this, the overhead is memory consumption of ~72 * memory(BlockInfo), which is relatively low. And it can even handle that `ba.l1.InfoAndTxsByHash(ctx, blockHash)` cache miss situation.

Reply:
> I think it's better to leave this, the overhead is memory consumption of ~72 * memory(BlockInfo), which is relatively low. And it can even handle that `ba.l1.InfoAndTxsByHash(ctx, blockHash)` cache miss situation.

yes, I think the cache is helpful.